### ISSUE TITLE
Turbo visits don't set `:target` to the targeted element

### ISSUE THREAD

**Issue by manuelpuyol** [hotwired/turbo#592]
When doing a Turbo visit to a link that has an anchor `#`, the targeted element does not have the `:target` pseudo-class.

### Steps to reproduce

1. In the Turbo project, visit http://localhost:9000/src/tests/fixtures/navigation.html
2. Search for elements with `:target` pseudo-class: `$(':target')` returns `null`
3. Reload page
4. Search for elements with `:target` pseudo-class: `$(':target')` returns element with id `element-id`

### Expected behavior

After clicking a link with an anchor, it's expected that the element which matches the URL hash has the `:target` pseudo-class.

This is important for apps that use that selector to add an indicator to the element - like navigating to a specific comment in a GitHub Issue:

![image](https://user-images.githubusercontent.com/11280312/169894341-f37da035-d64f-4d5f-a286-220d207cb9e0.png)

We are happy to look into this if we can get some pointers on where hashes may be getting interfered with :)

**Comment by domchristie** [comment]
This might be related to https://github.com/hotwired/turbo/issues/539.

I'm not sure there is a way to manually set the `:target` element, and unfortunately browsers don't seem to update it after `pushState`/`replaceState`. [Turbo does focus the target element](https://github.com/hotwired/turbo/blob/daabebb0575fffbae1b2582dc458967cd638e899/src/core/view.ts#L52-L62), so you might want to experiment with that.

Another option might be to create a link element behind-the-scenes with an `href` corresponding to the target element and with `data-turbo=false`, and then call `click` on it to mimic a non-Turbo navigation. (I've not tried this, but it might work.)

**Comment by domchristie** [comment]
> Another option might be to create a link element behind-the-scenes

This seems to work:

```js
addEventListener('turbo:load', target)
addEventListener('hashchange', target) // for same-page navigations

function target () {
  if (location.hash) {
    const a = document.createElement('a')
    a.href = `#${location.hash.slice(1)}`
    a.click()
  }
}
```

**Comment by manuelpuyol** [comment]
Wouldn't that cause a full page navigation after the Turbo visit?

**Comment by domchristie** [comment]
I don't think it will because the link is not contained in the document so Turbo won't handle the click.

**Comment by straight-shoota** [comment]
I've encountered this problem as well when `:target` CSS selectors stopped working with turbo navigation.
The workaround triggering an on-page navigation in the browser seems to work well (https://github.com/hotwired/turbo/issues/592#issuecomment-1137827028).

I think it would be great if turbo provided this functionality directly. It's supposed to replace native browser navigation, so it should also cover all facets of that.

**Comment by senny** [comment]
I'm facing the same issue but the fix from https://github.com/hotwired/turbo/issues/592#issuecomment-1137827028 does not fully fix it. I'm using anchors to link to a page with a large table. Scrolling to the target element works as expected. If I add the proposed fix, it does work with the CSS `:target` selector but it does scroll too far and the linked element is barely visible anymore.
